### PR TITLE
test(providers/microsoft): add test case for AzureIdentityCredentialAdapter.signed_session

### PR DESCRIPTION
**Pull request by Lee-W**
~~As discussed in [this stackoverflow thread](https://stackoverflow.com/questions/63384092/exception-attributeerror-defaultazurecredential-object-has-no-attribute-sig), part of the lib does not support azure-identity well and need this helper class.~~

~~This PR is dependency of https://github.com/apache/airflow/pull/33433, https://github.com/apache/airflow/pull/33469, https://github.com/apache/airflow/pull/33467~~

Add test case to signed_session which is the main function of this wrapper class

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/utils.py [hunk 1]
```diff
@@ -43,3 +49,47 @@ def get_field(*, conn_id: str, conn_type: str, extras: dict, field_name: str):
...
     return ret
+
+
+class AzureIdentityCredentialAdapter(BasicTokenAuthentication):
+    """Adapt azure-identity credentials for backward compatibility.
+
+    Adapt credentials from azure-identity to be compatible with SD
+    that needs msrestazure or azure.common.credentials
+
+    Check https://stackoverflow.com/questions/63384092/exception-attributeerror-defaultazurecredential-object-has-no-attribute-sig
```
Comment (uranusjr) [inline airflow/providers/microsoft/azure/utils.py]:
```suggestion
    Check https://stackoverflow.com/questions/63384092
```

**Comment by uranusjr** [comment]
Is it possible to test this?

**Comment by Lee-W** [comment]
> Is it possible to test this?

I've run it with https://github.com/apache/airflow/pull/33433 and succeeded. As for unit testing, I'm not that sure. Even if I add some, they might be mocking without checking anything. Is there any suggestions>

**Comment by uranusjr** [comment]
Not really, my impression is this wouldn’t be easy to test anyway, hence the question 😛

**Comment by potiuk** [comment]
> Is it possible to test this?
> Not really, my impression is this wouldn’t be easy to test anyway, hence the question 😛

Yes. We often do a mock-only tests and It's also important. Mock is really a representation of "How the author imagined the code would interact with the external world". Since you tested it, and you have good understanding how it works. Coding that in the form of mock is capturing that knowledge. 

Imagine somene else making a change to that code 2 months from now. Or even YOURSELF doing the 2 months from now (I always try to picture future-self coming back to the code after I completely forgot what it does).  If there is no mock and someone changes the behaviour accidentally, they won't notice and it might stop working accidentally.  But when it fails, you will look at your test, and you will be able to quickly recall the intent you had without heavy debugging.

I really treat those tests with mocks often as description  of what I thought the class in question shoudl be doing - so that I even don't have to remember it.

**Comment by Lee-W** [comment]
> > Is it possible to test this?
> > Not really, my impression is this wouldn’t be easy to test anyway, hence the question 😛
> 
> Yes. We often do a mock-only tests and It's also important. Mock is really a representation of "How the author imagined the code would interact with the external world". Since you tested it, and you have good understanding how it works. Coding that in the form of mock is capturing that knowledge.
> 
> Imagine somene else making a change to that code 2 months from now. Or even YOURSELF doing it 2 months from now (I always try to picture future-self coming back to the code after I completely forgot what it does). If there is no mock and someone changes the behaviour accidentally, they won't notice and it might stop working accidentally. But when it fails, you will look at your test, and you will be able to quickly recall the intent you had without heavy debugging.
> 
> I really treat those tests with mocks often as description of what I thought the class in question shoudl be doing - so that I even don't have to remember it.

Got it. Thanks @potiuk for such a detailed response. I'll try to add some mock tests tomorrow

**Comment by Lee-W** [comment]
Hi @potiuk @uranusjr , I tried to add a test in this PR. The original class is accidentally merged into the main branch in another PR. Could you please check whether such test looks good. Thanks!

**Review by potiuk — APPROVED** [review]
LGTM

**Pull request merged**